1. Import the numpy package under the name np 

In [1]:
import numpy as np

2. Create a null vector of size 20

In [8]:
a = np.zeros(20)


array([0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0.,
       0., 0., 0.])

3. Create a null vector of size 10 but the second value which is 1 

In [11]:
a= np.ones(20,dtype=int)
a

array([1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1])

4. Create a vector with values ranging from 10 to 49

In [14]:
a=np. arange(10,49)
a

array([10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 25, 26,
       27, 28, 29, 30, 31, 32, 33, 34, 35, 36, 37, 38, 39, 40, 41, 42, 43,
       44, 45, 46, 47, 48])

5 Reverse a vector (first element becomes last

In [21]:
a=np.arange(-49,-10)
a

array([-49, -48, -47, -46, -45, -44, -43, -42, -41, -40, -39, -38, -37,
       -36, -35, -34, -33, -32, -31, -30, -29, -28, -27, -26, -25, -24,
       -23, -22, -21, -20, -19, -18, -17, -16, -15, -14, -13, -12, -11])

6 Create a 3x3 matrix with values ranging from 0 to 8

In [23]:
a= np.arange(0,8)*3
a

array([ 0,  3,  6,  9, 12, 15, 18, 21])

7 Create a 3x3x3 array with random values 

In [44]:
a=np.random.rand(3,3)*3
a

array([[2.73890138, 2.28349888, 1.19993028],
       [1.73656367, 2.16305566, 1.06387619],
       [0.55794535, 2.04711905, 0.2023915 ]])

8. Create a 10x10 array with random values and find the minimum and maximum values


In [51]:
a= np.random.randint (10,20,size = (10,10))
a

array([[17, 14, 16, 18, 12, 19, 19, 12, 15, 12],
       [17, 15, 10, 10, 14, 16, 12, 11, 18, 17],
       [10, 11, 10, 17, 19, 17, 15, 18, 10, 11],
       [14, 14, 19, 15, 18, 18, 11, 10, 15, 10],
       [15, 18, 16, 17, 14, 16, 12, 16, 10, 18],
       [15, 10, 11, 13, 17, 15, 10, 11, 17, 16],
       [13, 16, 12, 13, 13, 11, 12, 14, 14, 14],
       [17, 11, 11, 16, 15, 15, 18, 19, 19, 13],
       [12, 14, 13, 16, 11, 16, 13, 12, 18, 11],
       [17, 17, 16, 19, 13, 10, 14, 16, 11, 19]])

In [52]:
np.min(a)

10

In [53]:
np.max(a)

19

9 Create a 5x5 matrix with row values ranging from 0 to 4 

In [55]:
a= np.random.randint(0,4,size = (5,5))
a

array([[3, 1, 3, 3, 1],
       [3, 0, 3, 2, 0],
       [1, 3, 0, 1, 2],
       [1, 1, 2, 2, 2],
       [0, 0, 2, 2, 1]])

10 Create a vector of size 10 with values ranging from 0 to 1, both excluded

In [58]:
a = np.random.rand(10,2)
a

array([[0.88428606, 0.3601945 ],
       [0.16571999, 0.89445875],
       [0.9470452 , 0.81671038],
       [0.57960224, 0.9340217 ],
       [0.73707741, 0.93376291],
       [0.34502443, 0.0050375 ],
       [0.49427277, 0.13316766],
       [0.19340808, 0.08095706],
       [0.60966654, 0.62389417],
       [0.32761972, 0.14459435]])

11. Create a random vector of size 10 and sort it 


In [80]:
a = np.random.randint(1,10, size=(1,10))
a

array([[3, 9, 8, 1, 1, 7, 6, 6, 2, 4]])

In [85]:
np.sort(a)

array([[1, 1, 2, 3, 4, 6, 6, 7, 8, 9]])

12 How to convert a float (32 bits) array into an integer (32 bits) in place?

In [89]:
a= np.array([5.5,3.6,4.8,5.9])
a.dtype

dtype('float64')

In [92]:
a= a.astype(int)
a.dtype
a

array([5, 3, 4, 5])

13 How to swap two rows of an array?

14 How to find the most frequent value in an array?


In [97]:
a =np. random.randint(10,20,size = (1,10))
a

array([[16, 16, 13, 16, 17, 10, 19, 17, 18, 12]])

In [106]:
import pandas as pd
a = pd.DataFrame(a)
a

0   1   2   3   4   5   6   7   8   9
0  16  16  13  16  17  10  19  17  18  12

15  How to reshape an array?

In [118]:
a1 = a.reshape(5,2)
a1

AttributeError: 'DataFrame' object has no attribute 'reshape'

16. Write a NumPy program to add, subtract, multiply, divide arguments element-wise.

val1: 7
val2: 3

17 Write a NumPy program to calculate the absolute value element-wise.

[-10.2, 122.2, -.20]

18 Write a NumPy program to round elements of the array to the nearest integer.

[-.7, -1.5, -1.7, 0.3, 1.5, 1.8, 2.0]   

19 Write a NumPy program to compute the mean, average of given array .

[0 1 2 3 4 5]

20 Write a NumPy program to compute  standard deviation, and variance of a given array .

[2,4,6,7,8]